### DISCUSSION TITLE
Quasar dev server and `quasar build --debug` works, but `quasar build` make broken bundle

### DISCUSSION THREAD

**Discussion by FragsterAt** [quasarframework/quasar#16304] [General - Components / Directives / etc] [answered]
I sure, that this is my fault, but i cant get, why it didnt work (

i make simple UI lib and try to integrate it in quasar app. here is a demo: https://github.com/FragsterAt/mdi-interface-demo
with `quasar dev` or `quasar build --debug`  it works fine, but with `quasar build` it fails, when I click on 'Open image' button.

Can anybody help me with it?

**Comment by rstoenescu** [comment, marked as answer]
`__name` and `__file` are only added for dev by Vue SFC (they are being used for HMR among other things). However, they are not included when building for production.
You need to rethink the strategy for your openView() function.

**Reply by rstoenescu** [reply]
Also, I'd add that relying on "private" methods or props is NOT a good strategy for any use case, regardless of the lib/framework you are using. These are generally not documented and can change without notice.

**Comment by FragsterAt** [comment]
> `__name` and `__file` are only added for dev by Vue SFC (they are being used for HMR among other things). However, they are not included when building for production. You need to rethink the strategy for your openView() function.

The only way out is to pre-register possible "names" like routes in vue-router or there is some more automatic way?

**Reply by rstoenescu** [reply]
You could inject your own options (use defineOptions) otherwise determine based on route.

**Comment by FragsterAt** [comment]
Completed lib published: https://www.npmjs.com/package/vue-multi-view
Thanks @rstoenescu for help!
Any feedback welcome!
